sonic-net/sonic-utilities
# [chassis] Add chassis support for CLI "config qos reload"

Fix issue: https://github.com/sonic-net/sonic-utilities/issues/3232

command "sudo config qos reload --ports Ethernetxx" fail on any chassis.
The difference comes because chassis config db has more entries:
e.g.
```
admin@chassis-lc1-1:/etc/sonic$ sudo grep -ir "ethernet88|0" config*
config_db0.json:        "Ethernet88|0": {
config_db0.json:        "lc1-1|asic0|Ethernet88|0-2": {
config_db0.json:        "lc2-1|asic0|Ethernet88|0-2": {
config_db0.json:        "lc3-1|asic0|Ethernet88|0-2": {
config_db0.json:        "lc1-1|asic0|Ethernet88|0": {
...
```
while a single-dut only has such key in the table:
```
admin@8111-06:/etc/sonic$ sudo grep -ir "ethernet88|0" config*
        "Ethernet88|0": {
        "Ethernet88|0-2": {
        "Ethernet88|0": {
```


#### What I did

#### How I did it

#### How to verify it
Before:
```
admin@chassis-lc7-1:~$ sudo config qos reload --ports Ethernet80
Traceback (most recent call last):
  File "/usr/local/bin/config", line 8, in <module>
    sys.exit(config())
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.9/dist-packages/config/main.py", line 2518, in reload
    _qos_update_ports(ctx, ports, dry_run, json_data)
  File "/usr/local/lib/python3.9/dist-packages/config/main.py", line 2659, in _qos_update_ports
    port, _ = key
ValueError: too many values to unpack (expected 2)
```

After:
command pass without any error.
In the device's config db, the entry for corresponding port was set to empty after command run:

**(Pdb) config_db.get_entry(table_name, '|'.join(key))
{'profile': 'ingress_lossy_profile'}**
(Pdb) '|'.join(key)
'Ethernet80|0'
(Pdb) key
('Ethernet80', '0')
(Pdb) table_name
'BUFFER_PG'
(Pdb) n
> /usr/local/lib/python3.9/dist-packages/config/main.py(2658)_qos_update_ports()
-> for key in entries:
**(Pdb)  config_db.get_entry(table_name, '|'.join(key))
{}**


#### Previous command output (if the output of a command-line utility has changed)

#### New command output (if the output of a command-line utility has changed)
